### DISCUSSION TITLE
Changing IED classes

### DISCUSSION THREAD

**Discussion by XDSenDX** [Vdauphin/HeartsAndMinds#1554] [Question and answer]
Hello is it possible to change H&M IED with others? I'm having a mod with custom IED that i want to use.

Cheers.

**Comment by Vdauphin** [comment]
Hello,
https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/mission.sqf#L413
Cheers

**Reply by XDSenDX** [reply]
Hello i'm putting the IED classes in this mod https://steamcommunity.com/sharedfiles/filedetails/?id=3048818056 but it seems not working. Tried on SP and the IED doesn't spawn, the marker on the map is also for the old IED.
Here's all my classes `btc_type_ieds_ace = ["iedd_ied_Cardboard", "iedd_ied_Cinder", "iedd_ied_CanisterFuel", "iedd_ied_Metal", "iedd_ied_Barrel", "iedd_ied_Barrel_Grey", ",iedd_ied_CanisterPlastic"];`

Cheers.

**Reply by Vdauphin** [reply]
If what you did is correct, it should work
You may not see the IED because is under the ground, just change the setting https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/param.hpp#L254-L259 to very easy

**Reply by XDSenDX** [reply]
Changed to very easy, still not working.

**Reply by Vdauphin** [reply]
How do you know it is not working?

**Reply by XDSenDX** [reply]
> How do you know it is not working?
> 

I got close to the ied marker and i don't see the model/ tried using mine detector which is still unable to detect any even i'm really close.

**Reply by Vdauphin** [reply]
Check your .rpt for errors and fix them

**Reply by XDSenDX** [reply]
`
22:23:04 (0:01:16) fnc\ied\create - [[2098.35,9433.31,9.53674e-007],"a3\structures_f\civ\constructions\woodenbox_f.p3d",148.191,t

22:23:04 rue]

22:23:04 Warning Message: No entry 'bin\config.bin/CfgVehicles/iedd_ied_Barrel.ammo'.

22:23:04 Warning Message: '/' is not a value

22:23:04 (0:01:16) fnc\ied\create - [[2165.29,9460.25,0],"a3\structures_f_heli\civ\constructions\gastank_01_khaki_f.p3d",277.5,fa

22:23:04 lse]

22:23:04 (0:01:16) fnc\ied\create - [[2392.26,9335.15,0],"a3\structures_f\civ\constructions\woodenbox_f.p3d",76.5117,false]

22:23:04 (0:01:16) fnc\ied\create - [[2103.77,9227.67,0],"a3\structures_f_epb\civ\garbage\garbagecontainer_closed_f.p3d",249.262,

22:23:04 false]`

Something is off

**Reply by XDSenDX** [reply]
Is there anyways to remove the IED structures?

**Reply by Vdauphin** [reply]
Your classname are wrong

Take a look here: https://github.com/Vdauphin/HeartsAndMinds/discussions/1393

**Comment by XDSenDX** [comment]
> Your classname are wrong
> 
> Take a look here: #1393

Nope i checked again the classnames are all correct, the name is literally iedd, which stands for IED disposal mod.

**Reply by Vdauphin** [reply]
There are not, if it was the case you won't see error in your .rpt
You need: type: [String](https://community.bistudio.com/wiki/String) - mine class name from CfgVehicles https://community.bistudio.com/wiki/createMine

**Reply by XDSenDX** [reply]
Ah i thought you were talking about typos.

I'm checking rn.

**Reply by XDSenDX** [reply]
Still can't find the issue here, i've checked everything, the "type:String" are all correct.

**Reply by Vdauphin** [reply]
Then something is wrong with the mod, may be ask the author how to spawn IED with https://community.bistudio.com/wiki/createMine

**Reply by XDSenDX** [reply]
> No, because our IEDs aren't mines.
>So I do not know is there way to modify h&m, I assume you are using some module to spawn random mines/explosives around area of it.
>If you know how to script you could do compat for h&m where you add check if type of "mine" is from your IEDs classes and it create with createVehicle command current IED else using createMine

This is their reply,
I wonder if doing this is possible?
Cheers.

**Reply by Vdauphin** [reply]
The A3 wiki explain everything:
This command creates objects of the CfgAmmo class named in [configFile](https://community.bistudio.com/wiki/configFile) [>>](https://community.bistudio.com/wiki/config_greater_greater_name) "CfgVehicles" [>>](https://community.bistudio.com/wiki/config_greater_greater_name) _type [>>](https://community.bistudio.com/wiki/config_greater_greater_name) "ammo".

If it is not an ammo you can't use createMine to spawn them. 

This is the way:
> If you know how to script you could do compat for h&m where you add check if type of "mine" is from your IEDs classes and it create with createVehicle command current IED else using createMine

**Comment by OverlordZorn** [comment]
i did an implementation of sth like that a long while ago, but its rather involved.

https://github.com/PulsarNeutronStar/HeartsAndMinds/tree/master/CVO_co%4030_Hearts_and_Minds.Altis/cvo/IED

Dont remember where i had to change some of the core functions to make this work, but this might still give you some insight/ideas.
